### PR TITLE
nixos/pam: refactor U2F, docs about u2f_keys path

### PR DESCRIPTION
**Pull request by kalbasit**
###### Motivation for this change

* change enableU2F option to u2f.* set
* add few u2f options (not all) to customize pam-u2f module
* document default u2f_keys locations

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] ~~macOS~~
   - [ ] ~~other Linux distributions~~
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Replaces #40455 and #11886.

### CODE REVIEW

**Comment by kalbasit** [comment]
@GrahamcOfBorg test pam-u2f

**Comment by kalbasit** [comment]
@spinus this PR is heavily based on your submission #11886. Please let me know if there's anything else you would like to add.
@Infinisil I've been using this PR for a few days now, and I have not had any issues, Do you have some time for a review? I will wait at least another week before merging to get a chance for more testing.

**Review by infinisil — CHANGES_REQUESTED** [review]

**Comment by kalbasit** [comment]
@Infinisil I've updated the PR and addressed all of your comments, PTAL.

File: nixos/modules/security/pam.nix
Comment (Infinisil) [inline nixos/modules/security/pam.nix]: Nitpick: `authfile` -> `authFile`

File: nixos/modules/security/pam.nix
Comment (Infinisil) [inline nixos/modules/security/pam.nix]: As per the other comments: `<literal>authfile</literal>` -> `<option>authFile</option>`

**Review by infinisil — CHANGES_REQUESTED** [review]

**Comment by kalbasit** [comment]
@Infinisil PTAL.

**Review by infinisil — APPROVED** [review]

**Comment by Infinisil** [comment]
Commit message should be changed to have the `nixos/...:` prefix. Feel free to merge this after that :)

**Comment by spinus** [comment]
@kalbasit thank you for pushing that forward!

**Pull request merged**
